From 281f831c95e45fa8384a3d9de1e62e242a3e2711 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:46 -0400
Subject: [cpufreq] acpi-cpufreq: Add comment under ACPI_ADR_SPACE_SYSTEM_IO case

Message-id: <1409235879-1283-73-git-send-email-prarit@redhat.com>
Patchwork-id: 89392
O-Subject: [RHEL7.1 PATCH BZ 1134639 072/185] acpi-cpufreq: Add comment under ACPI_ADR_SPACE_SYSTEM_IO case
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 1bab64d50e09706ee2e634b2df3c53346f0a0dbb
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Wed Oct 16 23:58:10 2013 +0200

    acpi-cpufreq: Add comment under ACPI_ADR_SPACE_SYSTEM_IO case

    policy->cur is now set by cpufreq core when cpufreq_driver->get() is defined and
    so drivers aren't required to set it. When space_id is ACPI_ADR_SPACE_SYSTEM_IO
    for acpi cpufreq driver it doesn't set ->get to a valid function pointer and so
    policy->cur is required to be set by driver.

    This is already followed in acpi-cpufreq driver. This patch adds a comment
    describing why we need to set policy->cur from driver.

    Suggested-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/acpi-cpufreq.c b/drivers/cpufreq/acpi-cpufreq.c
index 0d3589b..e8b0e8e 100644
--- a/drivers/cpufreq/acpi-cpufreq.c
+++ b/drivers/cpufreq/acpi-cpufreq.c
@@ -837,7 +837,12 @@ static int acpi_cpufreq_cpu_init(struct cpufreq_policy *policy)
 
  switch (perf->control_register.space_id) {
  case ACPI_ADR_SPACE_SYSTEM_IO:
-  /* Current speed is unknown and not detectable by IO port */
+  /*
+   * The core will not set policy->cur, because
+   * cpufreq_driver->get is NULL, so we need to set it here.
+   * However, we have to guess it, because the current speed is
+   * unknown and not detectable via IO ports.
+   */
   policy->cur = acpi_cpufreq_guess_freq(data, policy->cpu);
   break;
  case ACPI_ADR_SPACE_FIXED_HARDWARE:
-- 
1.7.1